UnityProject/Assets/Scripts/Objects/Kitchen/InteractableBarrel.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using Chemistry.Components;
+using Items;
+
+namespace Objects.Kitchen
+{
+	/// <summary>
+	/// Allows Microwave to be interacted with. Player can put food in the microwave to cook it.
+	/// The microwave can be interacted with to, for example, check the remaining time.
+	/// </summary>
+	[RequireComponent(typeof(FermentingBarrel))]
+	public class InteractableBarrel : MonoBehaviour, IExaminable, ICheckedInteractable<HandApply>,
+			IRightClickable, ICheckedInteractable<ContextMenuApply>
+	{
+		private FermentingBarrel barrel;
+
+		private void Start()
+		{
+			barrel = GetComponent<FermentingBarrel>();
+		}
+		public string Examine(Vector3 worldPos = default)
+		{
+			return $"It is currently {(barrel.Closed ? "closed, meaning the items inside are fermenting." : "open, meaning you can put things inside.")}";
```

Comment:
What if there're no items inside? Might give a false impression.

Reply:
true
i'll add a message to show if there's something in it or not

Reply:
while i'm at it, should i add the bubbling sound it makes on /tg/ when you ferment something?